scripts/check-buildinfo

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+"""
+Check newly added packages against their buildinfo files
+
+Example:
+    ./check-buildinfo package.deb package.buildinfo
+
+"""
+import argparse
+import hashlib
+import subprocess
+import sys
+from pathlib import Path
+from typing import List
+
+from debian.deb822 import BuildInfo
+
+
+def lookup_buildinfos(buildinfos: Path) -> dict:
+    """Extract checksums out of every buildinfo file we can find"""
+    data = {}
+    for path in buildinfos.glob("**/*.buildinfo"):
+        info = BuildInfo(path.read_text())
+        for details in info['Checksums-Sha256']:
+            if details['name'].endswith('.deb'):
+                data[details['name']] = details['sha256']
```

Comment:
Would it be a useful sanity-check to make sure we haven't already seen a checksum for `details['name']`?

```suggestion
            if details['name'].endswith('.deb'):
                if details['name'] in data:
                    raise ValueError(f"ERROR: duplicate checksum for {details['name']}")
                data[details['name']] = details['sha256']
```

Reply:
I think that's a good idea just as general hardening.

In theory people will commit the buildinfo files using the generated filename, so there shouldn't be room for duplicates, but it's also possible we'll move packages around (e.g. securedrop-grsec was formerly built by securedrop server, now it's built by the kernel stuff) and accidentally cause duplicates. An attacker could theoretically also add a buildinfo with an easy-to-overlook extra entry that clobbers the correct checksum with their own...so overall, good idea!